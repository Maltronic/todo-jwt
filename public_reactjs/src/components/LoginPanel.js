import PropTypes from 'prop-types';
import React, {Component} from 'react';

export default class LoginPanel extends Component {
    constructor(props) {
        super(props);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.state = {
            username: '',
            password: ''
        };
    }

    handleSubmit = (event) => {
        event.preventDefault();
        return this.props.submitCallback(this.state.username, this.state.password);
    };

    handleChange = (event) => {
        this.setState({[event.target.name]: event.target.value});
    };

    render() {
        const content = (this.props.completedPhrase) ? this.displayCompleted() : this.displayForm();
        return (
            <form onSubmit={this.handleSubmit}
                  className="form-signin pull-right">
                <div><h4>{this.props.title}</h4></div>
                {content}
            </form>
        );
    }

    displayCompleted = () => (
        <div>{this.props.completedPhrase}</div>
    );

    displayForm = () => (
        <div>
            <div>
                <label>User name:</label><br/>
                <input type="text" name="username" placeholder="username..." value={this.state.username}
                       onChange={this.handleChange}/>
            </div>
            <div>
                <label>Password:</label><br/>
                <input type="password" name="password" placeholder="password..." value={this.state.password}
                       onChange={this.handleChange}/>
            </div>
            <input className="btn btn-primary" type="submit" value="submit"/>
        </div>
    );
}

LoginPanel.propTypes = {
    completedPhrase: PropTypes.string,
    submitCallback: PropTypes.func.isRequired,
    title: PropTypes.string
};
