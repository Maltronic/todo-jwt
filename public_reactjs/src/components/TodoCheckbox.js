import PropTypes from 'prop-types';
import React, { Component } from 'react';

export default class TodoCheckbox extends Component {
    render() {
        return (<input type="checkbox" checked={this.props.todo.done} onChange={() => {
            this.props.toggleCallback(this.props.todo)
        }}/>);
    }
}

TodoCheckbox.propTypes = {
    todo: PropTypes.object.isRequired,
    toggleCallback: PropTypes.func.isRequired,
};
