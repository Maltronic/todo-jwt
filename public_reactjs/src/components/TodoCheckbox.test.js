import React from 'react';
import { shallow, mount } from 'enzyme';
import LoginPanel from "./LoginPanel";
import TodoCheckbox from "./TodoCheckbox";

describe('TodoCheckbox renders correctly', () => {
    let props = {
        todo: {
            __v: 0,
            _id: "5814eaee84f3bf7ff597e16f",
            createdAt: "2016-10-29T18:31:10.075Z",

            done: true,
            priority: 0,
            text: "test"
        },
        toggleCallback: null
};

    beforeEach(() => {
        props.toggleCallback = jest.fn();
    });

    it('Matches Snapshot', () => {
        const wrapper = shallow(
            <TodoCheckbox {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("Can be submitted", () => {
        const wrapper = mount(<TodoCheckbox {...props} />);
        wrapper.simulate('change');
        expect(props.toggleCallback).toHaveBeenCalledTimes(1);
    });
});

