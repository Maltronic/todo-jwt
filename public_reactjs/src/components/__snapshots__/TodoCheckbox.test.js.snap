// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoCheckbox renders correctly Matches Snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
/>,
  "nodes": Array [
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TodoCheckbox
        todo={
                Object {
                        "__v": 0,
                        "_id": "5814eaee84f3bf7ff597e16f",
                        "createdAt": "2016-10-29T18:31:10.075Z",
                        "done": true,
                        "priority": 0,
                        "text": "test",
                      }
        }
        toggleCallback={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TodoCheckbox {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "todo": Object {
            "__v": 0,
            "_id": "5814eaee84f3bf7ff597e16f",
            "createdAt": "2016-10-29T18:31:10.075Z",
            "done": true,
            "priority": 0,
            "text": "test",
          },
          "toggleCallback": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
/>,
        "_debugID": 2,
        "_renderedOutput": <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TodoCheckbox
    todo={
        Object {
            "__v": 0,
            "_id": "5814eaee84f3bf7ff597e16f",
            "createdAt": "2016-10-29T18:31:10.075Z",
            "done": true,
            "priority": 0,
            "text": "test",
          }
    }
    toggleCallback={[Function]}
/>,
}
`;
